dotnet/samples/Concepts/Filtering/TelemetryWithFilters.cs

```diff
@@ -167,34 +177,43 @@ public async Task OnFunctionInvocationAsync(FunctionInvocationContext context, F
             }
         }
 
-        private async Task LogFunctionResultAsync(FunctionInvocationContext context)
+        private void ProcessFunctionResult(FunctionResult functionResult)
+        {
+            string? result = functionResult.GetValue<string>();
+            object? usage = functionResult.Metadata?["Usage"];
+
+            if (!string.IsNullOrWhiteSpace(result))
+            {
+                logger.LogTrace("Function result: {Result}", result);
+            }
+
+            if (logger.IsEnabled(LogLevel.Information) && usage is not null)
+            {
+                logger.LogInformation("Usage: {Usage}", JsonSerializer.Serialize(usage));
+            }
+        }
+
+        private async IAsyncEnumerable<StreamingChatMessageContent> ProcessFunctionResultStreamingAsync(IAsyncEnumerable<StreamingChatMessageContent> data)
         {
-            string? result = null;
             object? usage = null;
 
-            if (context.IsStreaming)
+            var stringBuilder = new StringBuilder();
+
+            await foreach (var item in data)
             {
-                var stringBuilder = new StringBuilder();
+                yield return item;
 
-                await foreach (var item in context.Result.GetValue<IAsyncEnumerable<StreamingChatMessageContent>>()!)
+                if (item.Content is not null)
                 {
-                    if (item.Content is not null)
-                    {
-                        stringBuilder.Append(item.Content);
-                    }
-
-                    usage = item.Metadata?["Usage"];
+                    stringBuilder.Append(item.Content);
                 }
 
-                result = stringBuilder.ToString();
-            }
-            else
-            {
-                result = context.Result.GetValue<string>();
-                usage = context.Result.Metadata?["Usage"];
+                usage = item.Metadata?["Usage"];
```

Comment:
nit.

Usage in streaming comes in the last chunk, we dont need to set every iteration.

```suggestion
                usage ??= item.Metadata?["Usage"];
```

Reply:
Actually as you mentioned before, for setting values it doesn't work as I suggested. Thanks for double checking that. resolving the comment.